src/stax/ui.c

```diff
@@ -0,0 +1,583 @@
+/*******************************************************************************
+ *   Password Manager application
+ *   (c) 2017-2023 Ledger SAS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+
+#include <os.h>
+#include <string.h>
+
+#if defined(TARGET_STAX)
+
+#include <nbgl_layout.h>
+#include <nbgl_page.h>
+#include <nbgl_use_case.h>
+
+#include "../error.h"
+#include "../globals.h"
+#include "../metadata.h"
+#include "../options.h"
+#include "../password.h"
+#include "../dispatcher.h"
+
+#include "glyphs.h"
+#include "password_list.h"
+#include "ui.h"
+
+static nbgl_page_t *pageContext = NULL;
+static nbgl_layout_t *layoutContext = NULL;
+#define MAX_ERROR_MSG_SIZE 100
+static char errorMessage[MAX_ERROR_MSG_SIZE] = {0};
+
+enum {
+    BACK_BUTTON_TOKEN = FIRST_USER_TOKEN,
+    CREATE_TOKEN,
+    UPPERCASE_TOKEN,
+    LOWERCASE_TOKEN,
+    NUMBERS_TOKEN,
+    BARS_TOKEN,
+    EXT_SYMBOLS_TOKEN,
+    ERASE_METADATA_TOKEN,
+    NO_ENTER_TOKEN,
+    CHOICE_WRITE_TOKEN,
+    CHOICE_DISPLAY_TOKEN,
+    CHOICE_CREATE_TOKEN,
+    CHOICE_DELETE_TOKEN,
+    KBD_TEXT_TOKEN,
+};
+
+static void release_context(void) {
+    if (pageContext != NULL) {
+        nbgl_pageRelease(pageContext);
+        pageContext = NULL;
+    }
+    if (layoutContext != NULL) {
+        nbgl_layoutRelease(layoutContext);
+        layoutContext = NULL;
+    }
+}
+
+static void display_choice_page(void);
+static void display_create_pwd_page(void);
+static void display_settings_page(void);
+
+/*
+ * Error display
+ */
+static void display_error_page(error_type_t error) {
+    message_pair_t msg = get_error(error);
+    const size_t msg_length = strlen(msg.first) + strlen(msg.second) + 2;
+    snprintf(&errorMessage[0],
+             msg_length > MAX_ERROR_MSG_SIZE ? MAX_ERROR_MSG_SIZE : msg_length,
+             "%s\n%s",
+             (char *) PIC(msg.first),
+             (char *) PIC(msg.second));
+    nbgl_useCaseStatus(&errorMessage[0], false, &display_choice_page);
```

Comment:
Isn't it simpler as:
```suggestion
    snprintf(errorMessage,
             sizeof(errorMessage),
             "%s\n%s",
             (char *) PIC(msg.first),
             (char *) PIC(msg.second));
    nbgl_useCaseStatus(errorMessage, false, &display_choice_page);
```

Reply:
yep